### ISSUE TITLE
Teleport should reject edits to SSO connectors without secrets

### ISSUE THREAD

**Issue by klizhentas** [gravitational/teleport#28324]
Teleport should reject edits to SSO connectors with an empty or missing secrets.

If users run the following sequence:

```
tctl get sso/okta > okta.yaml
tctl create -f okta.yaml
```

They are guaranteed to break the SSO. To prevent this, the following should happen:

```
tctl get sso/okta > okta.yaml
tctl create -f okta.yaml
ERROR: This connector is missing required secrets. Please use 
tctl get sso/okta --with-secrets to fix the error
```

To capture the fact that connector is missing secrets, we can simply record this fact in the file itself when returning it with no secrets. We should also check for usual suspects - missing PAM section if it's required.

**Comment by webvictim** [comment]
Related ancient feature request: https://github.com/gravitational/teleport/issues/6815

**Comment by zmb3** [comment]
Interesting, we appear to already handle this case:

https://github.com/gravitational/teleport/blob/245e5d831e5d9557774e1b79603cd9bf033e3cc8/tool/tctl/common/resource_command.go#L784-L793

**Comment by zmb3** [comment]
@hugoShaka you fixed this, right?

**Comment by hugoShaka** [comment]
At least for SAML connectors. I don't know about OIDC ones.

- https://github.com/gravitational/teleport/pull/44655

**Comment by webvictim** [comment]
Yeah, the code for OIDC connectors has no similar handling for the client secret.

**Comment by zmb3** [comment]
```
➜ tctl get oidc > oidc.yaml
➜ tctl create -f oidc.yaml
ERROR: OIDC connector is missing required client_secret
```

Looks like I addressed it for OIDC as part of #46041
